.github/workflows/prod-upgrade-deploy-test.yml

```diff
@@ -86,9 +86,10 @@ jobs:
             ${{ runner.os }}-pip-${{ env.PROD_PYTHON_VERSION }}-${{ secrets.CACHE_SEED }}-
 
       - name: Install Python dependencies
+        # TODO: Remove `|| […]` once deployed to production
         run: |
-          python -m pip install --upgrade pip
-          python -m pip install poetry
+          python -m pip install --requirement=geostore/pip.txt || python -m pip install --upgrade pip
+          python -m pip install --requirement=geostore/poetry.txt || python -m pip install poetry
```

Comment:
Just wondering why `||` is needed here. Is there an expectation that one might fail and we fallback to the other method of installing pip and poetry? Would we expect any discrepancy between using the first installation method over the other? Mostly asking out of curiosity....

Reply:
There's a hint in the comment above:

> TODO: Remove `|| […]` once deployed to production

It's explained further in the commit message:

> fix: Fall back to old install method on old commit
>
> After checking out the current production commit we don't have the `.txt` lock files available.